### ISSUE TITLE
Bug: (getter-return: does not check Object.create )

### ISSUE THREAD

**Issue by hirasawayuki** [eslint/eslint#16419]
### Environment

Node version: v18.3.0
npm version: v8.19.2
Local ESLint version: v8.25.0 (Currently used)
Global ESLint version: Not found
Operating System: darwin 20.6.0

### What parser are you using?

Default (Espree)

### What did you do?

<details>
<summary>Configuration</summary>

```
<!-- Paste your configuration here -->
```
</details>

```js
Object.create(Object.prototype, {
  bar: {
    configurable: false,
    get: function() {}, // ← does not check
    set: function(value) {
      console.log('Setting `o.bar` to', value);
    }
  }
});
```

### What did you expect to happen?

warnings on getters which have no return values.

### What actually happened?

no warning on Object.create usage.

### Participation

- [X] I am willing to submit a pull request for this issue.

**Comment by nzakas** [comment]
I support this change.
